tools/HiveCompare/HiveCompare/Program.cs

```diff
@@ -0,0 +1,194 @@
+                FixVirtualSpaceAndPrefixSuffixes = true,
+                PersianShape = !noOptOption.HasValue()
+            };
+            try
+            {
+                var book = bookEngine.ProcessBook(inputFile.Value());
+                var outputPath = outputFile.HasValue() ? outputFile.Value() : null;
+                var savedPath = book.SaveAs();
+                if (outputPath != null)
+                {
+                    File.Copy(savedPath, outputPath, true);
+                    File.Delete(savedPath);
+                    savedPath = Path.GetFullPath(outputPath);
+                }
+                Console.WriteLine($"Congratulations, operation was successful, File saved at '{savedPath}'.");
+            }
+            catch (Base.Exceptions.EBookFormatNotSupported ex)
+            {
+                Console.WriteLine(ex.Message);
+                return 0;
+            }
+            catch (FileNotFoundException)
+            {
+                Console.WriteLine($"Could not find file '{inputFile.Value()}'.");
+                return 0;
+            }
+            catch (UnauthorizedAccessException)
+            {
+                Console.WriteLine($"File '{inputFile.Value()}' access denied.");
+                return 0;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine("Error, unfortunatelly something went wrong. You can check log file next to your epub file");
+                var logPath = FileHelper.LogFileName(inputFile.Value(), "errors.txt");
+                if (File.Exists(logPath))
+                    File.Delete(logPath);
+                do
+                {
+                    File.AppendAllText(logPath, ex.Message + "\r\n" + ex.StackTrace + "\r\n--------------------------\r\n");
+                    ex = ex.InnerException;
+                } while (ex != null);
+                return 0;
+            }
+            return 1;
+            
+        });
+        return cli;
+
+        
+    }
+
+    private static void ParseTests(string firstFile, string secondFile)
+    {
+        Dictionary<string, TestCase> testCases1;
+        Dictionary<string, TestCase> testCases2;
+        try
+        {
+            var file1 = File.ReadAllText(firstFile);
+            var file2 = File.ReadAllText(secondFile);
+
+            var hiveTest1 = JsonSerializer.Deserialize<HiveTestResult>(file1, SERIALIZER_OPTIONS);
+            var hiveTest2 = JsonSerializer.Deserialize<HiveTestResult>(file2, SERIALIZER_OPTIONS);
+
+            if (hiveTest1 is null || hiveTest2 is null)
+            {
+                Console.WriteLine("Could not parse one of the files!");
+                return;
+            }
+            testCases1 = hiveTest1.TestCases.Values.DistinctBy(v => v.Name + v.Description).ToDictionary(v => v.Name + v.Description);
+            testCases2 = hiveTest2.TestCases.Values.DistinctBy(v => v.Name + v.Description).ToDictionary(v => v.Name + v.Description);
```

Comment:
Shall we define `Key => Name + Description` ?

Reply:
I think it would be easier to have one Dictionary with <key, (Test1, Test2)>. Will be easier to compare results.

Reply:
not sure if doing that is worth it. We have two dictionaries that could be inconsistent meaning, one could have keys that arent in the second and vice versa. not sure merging them into one dictionary is computationally efficient.

Reply:
First comment addressed